Lib/importlib/metadata.py

```diff
@@ -0,0 +1,271 @@
+    module = import_module(path)
+    return getattr(module, name)
+
+
+def version(package: Package) -> str:
+    """Get the version string for the named package.
+
+    :param package: The module object for the package or the name of the
+        package as a string.
+    :return: The version string for the package as defined in the package's
+        "Version" metadata key.
+    """
+    return distribution(package).version
+
+
+class MetadataPathFinder:
+    """A degenerate finder for distribution packages on the file system.
+
+    This finder supplies only a find_distribution() method for versions
+    of Python that do not have a PathFinder find_distribution().
+    """
+    @staticmethod
+    def find_spec(*args, **kwargs):
+        return None
+
+    @classmethod
+    def find_distribution(cls, name):
+        paths = cls._search_paths(name)
+        dists = map(PathDistribution, paths)
+        return next(dists, None)
+
+    @classmethod
+    def _search_paths(cls, name):
+        """
+        Find metadata directories in sys.path heuristically.
+        """
+        return itertools.chain.from_iterable(
+            cls._search_path(path, name)
+            for path in map(Path, sys.path)
+            )
+
+    @classmethod
+    def _search_path(cls, root, name):
+        if not root.is_dir():
+            return ()
+        return (
+            item
+            for item in root.iterdir()
+            if item.is_dir()
+            and str(item.name).startswith(name)
+            and re.match(rf'{name}(-.*)?\.(dist|egg)-info', str(item.name))
+            )
+
+
+class PathDistribution(Distribution):
+    def __init__(self, path):
+        """Construct a distribution from a path to the metadata directory."""
+        self._path = path
+
+    def load_metadata(self, name):
+        """Attempt to load metadata given by the name.
+
+        :param name: The name of the distribution package.
+        :return: The metadata string if found, otherwise None.
+        """
+        filename = os.path.join(self._path, name)
+        with contextlib.suppress(FileNotFoundError):
+            with open(filename, encoding='utf-8') as fp:
+                return fp.read()
+        return None
+
+
+class WheelMetadataFinder:
+    """A degenerate finder for distribution packages in wheels.
+
+    This finder supplies only a find_distribution() method for versions
+    of Python that do not have a PathFinder find_distribution().
+    """
+    @staticmethod
+    def find_spec(*args, **kwargs):
+        return None
+
+    @classmethod
+    def find_distribution(cls, name):
+        try:
+            module = import_module(name)
+        except ImportError:
+            return None
+        archive = getattr(module.__loader__, 'archive', None)
+        if archive is None:
+            return None
+        try:
+            name, version = Path(archive).name.split('-')[0:2]
+        except ValueError:
```

Comment:
What are you expecting to raise `ValueError`? The `Path` call?

Reply:
This code is stale.